lib/matplotlib/axes/_axes.py

```diff
@@ -4393,6 +4395,36 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
         path = marker_obj.get_path().transformed(
             marker_obj.get_transform())
         if not marker_obj.is_filled():
+            if orig_edgecolor is not None:
+                cbook._warn_external(
+                    f"You passed a edgecolor/edgecolors ({orig_edgecolor!r}) "
+                    f"for an unfilled marker ({marker!r}).  Matplotlib is "
+                    "ignoring the edge color in favor of the facecolor.  This "
+                    "behavior may change in the future."
+                )
+            # We need to handle markers that can not be filled (like
+            # '+' and 'x') differently than markers that can be
+            # filled, but have their fillstyle set to 'none'.  This is
+            # to get:
+            #
+            #  - respecting the fillestyle if set
+            #  - maintaining back-compatibility for querying the facecolor of
+            #    the un-fillable markers.
+            #
+            # While not an ideal situation, but is better than the
+            # alternatives.
+            if marker_obj.get_fillstyle() == 'none':
```

Comment:
I'm still somewhat confused why we need special-casing in the drawing function. Could this not be handled in the `MarkerStyle` so that `MarkerStyle('o', fillstyle='none')` and `MarkerStyle('x')` have an identical behavior?